src/QMCDrivers/tests/test_DMCBatched.cpp

```diff
@@ -30,7 +30,7 @@ TEST_CASE("DMCBatched::calc_default_local_walkers", "[drivers]")
 
 
   auto testWRTWalkersPerRank = [&dtest](int walkers_per_rank) {
-                                 DMCBatched dmc_batched(std::move(dtest()));
```

Comment:
No need to fix here if it works now, but it is unsafe. When I was working on https://github.com/QMCPACK/qmcpack/pull/2221, I noticed the MCPopulation move constructor and move assign operator cannot be compiler generated by default because ParticleSet cannot be moved. The reason is that the compiler tries to use copy assign operator of ParticleSet which should be either deleted or implemented just like the copy constructor.

Reply:
Since most of this PR has been long exploration of the unintended consequences of "unsafe" things I'd like to just leave this fixed.